File: leetcode-ts/src/no0001.ts
export class No0001 {
    twoSum(nums: number[], target: number): number[] {
        let answer: number[];
        for (let i = 0; i < nums.length; i++) {
            for (let j = 0; j < nums.length; j++) {
                if ((nums[i] + nums[j]) === target) {
                    answer = [i, j];
                    break;
                }
            }
        }
        // @ts-ignore
        return answer;
    };
}

File: leetcode/src/main/java/com/github/lucaseo90/ds/day01/array/No0001.java
package com.github.lucaseo90.ds.day01.array;

import java.util.HashMap;

public class No0001 {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        int[] answer = null;
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(target - nums[i])) {
                answer = new int[]{map.get(target - nums[i]), i};
            }
            map.put(nums[i], i);
        }

        return answer;
    }
}
